src/Modules/OrchardCore.Commerce/Controllers/ShoppingCartController.cs

```diff
@@ -83,7 +88,16 @@ public async Task<ActionResult> Index(string shoppingCartId = null)
 
     [HttpGet]
     [Route("cart-empty")]
-    public IActionResult Empty() => View();
+    public async Task<ActionResult> Empty()
+    {
+        var trackingConsentFeature = HttpContext.Features.Get<ITrackingConsentFeature>();
+        if (trackingConsentFeature?.CanTrack == false)
+        {
+            await _notifier.ErrorAsync(H["You have to accept the cookies, to add items to your shopping cart!"]);
```

Comment:
In the error message, provide tips on what the user can do. E.g. disable privacy-related browser extensions for the current site.